Model/Api/CreateOrder.php

```diff
@@ -0,0 +1,314 @@
+
+    /**
+     * @var LogHelper
+     */
+    private $logHelper;
+
+    /**
+     * @var Request
+     */
+    private $request;
+
+    /**
+     * @var Bugsnag
+     */
+    private $bugsnag;
+
+    /**
+     * @var Response
+     */
+    private $response;
+
+    /**
+     * @var ConfigHelper
+     */
+    private $configHelper;
+
+    /**
+     * @var CartHelper
+     */
+    private $cartHelper;
+
+    /**
+     * @param HookHelper   $hookHelper
+     * @param OrderHelper  $orderHelper
+     * @param CartHelper   $cartHelper
+     * @param LogHelper    $logHelper
+     * @param Request      $request
+     * @param Bugsnag      $bugsnag
+     * @param Response     $response
+     * @param ConfigHelper $configHelper
+     */
+    public function __construct(
+        HookHelper $hookHelper,
+        OrderHelper $orderHelper,
+        CartHelper $cartHelper,
+        LogHelper $logHelper,
+        Request $request,
+        Bugsnag $bugsnag,
+        Response $response,
+        ConfigHelper $configHelper
+    ) {
+        $this->hookHelper = $hookHelper;
+        $this->orderHelper = $orderHelper;
+        $this->logHelper = $logHelper;
+        $this->request = $request;
+        $this->bugsnag = $bugsnag;
+        $this->response = $response;
+        $this->configHelper = $configHelper;
+        $this->cartHelper = $cartHelper;
+    }
+
+    /**
+     * Pre-Auth hook: Create order.
+     *
+     * @api
+     *
+     * @param string $type - "order.create"
+     * @param array $order
+     * @param string $currency
+     *
+     * @return void
+     * @throws \Exception
+     */
+    public function execute(
+        $type = null,
+        $order = null,
+        $currency = null
+    ) {
+        try {
+            if ($type !== 'order.create') {
+                throw new BoltException(
+                    __('Invalid hook type!'),
+                    null,
+                    self::E_BOLT_GENERAL_ERROR
+                );
+            }
+
+            $payload = $this->request->getContent();
+
+            $this->validateHook();
+
+            if (empty($order)) {
+                throw new BoltException(
+                    __('Missing order data.'),
+                    null,
+                    self::E_BOLT_GENERAL_ERROR
+                );
+            }
+
+            $quoteId = $this->getQuoteIdFromPayloadOrder($order);
+            /** @var \Magento\Quote\Model\Quote $quote */
+            $quote = $this->loadQuoteData($quoteId);
+
+            /** @var \Magento\Sales\Model\Order $createOrderData */
+            $createOrderData = $this->orderHelper->preAuthCreateOrder($quote, $payload);
+
+            $this->sendResponse(200, [
+                'status'    => 'success',
+                'message'   => 'Order create was successful',
+                'display_id' => $createOrderData->getId(),
+                'total'      => $createOrderData->getGrandTotal() * 100,
+                'order_received_url' => '',
```

Comment:
Is this url supposed to be blank?

Reply:
Nope. I am working on it with Niraj now.

Reply:
Fixed